File: 1-two-sum/two-sum.java
class Solution 
{
    public int[] twoSum(int[] nums, int target) 
    {
        for(int i = 0; i<nums.length; i++)
        {
            for(int j = nums.length-1; j>=0;j--)
            {
                if (nums[i] + nums[j]==target && i!=j)
                {
                    return new int[] {i,j};
                }
            }
        }
        return new int[] {};
    }
}

File: 1-two-sum/two-sum.scala
object Solution {
    def twoSum(nums: Array[Int], target: Int): Array[Int] = {
      val map = nums.zipWithIndex.toMap

      val r = nums.indices.indexWhere( i => map.get(target - nums(i)).exists(_ != i))
      if (r == -1) Array(-1, -1) else Array(r, map(target - nums(r)))
    }
}